### PR TITLE
feat: loading indicator

### LINKED ISSUE #512: Loading indicator in terminal during building
Would be nice if we can have a loading bar when the command starts as some project might be big to build that hangs for a while

### PR DESCRIPTION
**Pull request by hyoban**
continue #523
close #512

In this PR, I stop the spinner before new log.

### CODE REVIEW

**Review by huozhi — CHANGES_REQUESTED** [review]
Hi, thanks for the PR. It was intentionally to keep logging simple and zero-dependency to keep everything lightweight. I'd like to have a implementation built-in instead of relying on other package and changing all the logging styles

**Comment by hyoban** [comment]
Hi, I just reverted this PR to use ora, do we need to restore the spinner after the new log output?

File: src/logger.ts [hunk 1]
```diff
@@ -1,26 +1,44 @@
...
+
+function clearSpinner() {
+  if (buildingSpinner.isSpinning) {
+    buildingSpinner.stop()
+  }
+}
+
 export const logger = {
   log(...arg: any[]) {
+    clearSpinner()
```
Comment (huozhi) [inline src/logger.ts:23]: logging should be isolated from loading, otherwise every time you log sth it will accidentally stop the spinner

Comment (hyoban) [reply src/logger.ts:23]: Yes, so I asked do we need to restore the spinner after the new log output. It seems the only way we can make ora doesn't swallow stdout https://github.com/sindresorhus/ora/issues/90. This is also how vitejs does https://github.com/vitejs/vite/pull/269/files.

**Pull request closed without merging**